source/1_engine/rex_engine/src/diagnostics/assert.cpp

```diff
@@ -0,0 +1,87 @@
+#include "rex_engine/diagnostics/assert.h"
+
+#include "rex_engine/diagnostics/stacktrace.h"
+#include "rex_engine/diagnostics/logging/log_macros.h"
+#include "rex_engine/debug_types.h"
+
+namespace rex
+{
+  DECLARE_LOG_CATEGORY_EXTERN(LogAssert, LogVerbosity::Log);
+  DEFINE_LOG_CATEGORY(LogAssert);
+
+  DebugVector<AssertContext>& contexts()
+  {
+    thread_local DebugVector<AssertContext> contexts;
+    return contexts;
+  }
+
+  void rex_assert(const rsl::fmt_stack_string& msg)
```

Comment:
I'm going to need some explanation why this is such a complex thing.
Is there a reason why it does not log a message and breaks.